File: examples/scenarios/_example-template/snippets/react-native/cancelOnEvent.ts
// Example placement: target action completion handler in app JS/TS layer (pseudo-code).
// Adapt to: your event tracking layer and local notification cancellation integration.
// Shared config: examples/scenarios/_example-template/openclix.config.json
// Events used: first_action_completed
// Note: OpenClix APIs shown here are placeholders until the reference SDK is published.

type OpenClixManagerLike = {
  ingestEvent: (event: { name: string; appState: 'foreground' | 'background' }) => void;
};

export function cancelTemplateCampaignAfterFirstAction(openClix: OpenClixManagerLike): void {
  openClix.ingestEvent({
    name: 'first_action_completed',
    appState: 'foreground',
  });
}

File: examples/scenarios/_example-template/snippets/android/CancelOnEvent.kt
// Example placement: target action completion handler (pseudo-code).
// Adapt to: your analytics/event tracking integration and runtime store.
// Shared config: examples/scenarios/_example-template/openclix.config.json
// Events used: first_action_completed
// Note: OpenClix APIs shown here are placeholders until the reference SDK is published.

fun cancelTemplateCampaignAfterFirstAction(openClix: OpenClixManagerLike) {
    val event = OpenClixEventLike(
        name = "first_action_completed",
        appState = OpenClixAppStateLike.FOREGROUND
    )
    openClix.ingestEvent(event)
}
